extensions/micrometer/deployment/src/main/java/io/quarkus/micrometer/deployment/binder/KafkaStreamsBinderProcessor.java

```diff
@@ -0,0 +1,32 @@
+package io.quarkus.micrometer.deployment.binder;
+
+import java.util.function.BooleanSupplier;
+
+import io.quarkus.arc.deployment.AdditionalBeanBuildItem;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.micrometer.runtime.MicrometerRecorder;
+import io.quarkus.micrometer.runtime.config.MicrometerConfig;
+
+public class KafkaStreamsBinderProcessor {
+
+    static final String KAFKA_STREAMS_CLASS_NAME = "org.apache.kafka.streams.KafkaStreams";
+    static final Class<?> KAFKA_STREAMS_CLASS_CLASS = MicrometerRecorder.getClassForName(KAFKA_STREAMS_CLASS_NAME);
+
+    static final String KAFKA_STREAMS_METRICS_PRODUCER_CLASS_NAME = "io.quarkus.micrometer.runtime.binder.kafkastream.KafkaStreamsMetricsProducer";
+
+    static class KafkaStreamsSupportEnabled implements BooleanSupplier {
+        MicrometerConfig mConfig;
+
+        public boolean getAsBoolean() {
+            return KAFKA_STREAMS_CLASS_CLASS != null && mConfig.checkBinderEnabledWithDefault(mConfig.binder.kafka);
+        }
+    }
+
+    @BuildStep(onlyIf = KafkaStreamsBinderProcessor.KafkaStreamsSupportEnabled.class)
+    AdditionalBeanBuildItem createKafkaStreamsMetricsProducer() {
+        return AdditionalBeanBuildItem.builder()
+                .addBeanClass(KAFKA_STREAMS_METRICS_PRODUCER_CLASS_NAME)
+                .setUnremovable().build();
+    }
+
+}
```

Comment:
In thinking about this -- these (the new BooleanSupplier and BuildStep) can be placed into the existing Kafka processor to keep things together..

Reply:
Yes, that makes sense. I will move this processor and the producer into the kafka package.

Reply:
After looking at the logic a bit, I am not sure if this should be combined.

Yes, both deal with Kafka, but the Kafka classes come from two different dependencies (org.apache.kafka:kafka-clients and org.apache.kafka:kafka-streams)

This must be taken into account especially by the producer.

Reply:
I meant only to put them all in the same file given they are coming from the same general config (measure kafka or not). 
Preserve the test for unique class given the different dependency (e.g. preserve unique BooleanSupplier)

Reply:
I moved  BooleanSupplier and BuildStep into the KafkaBinderProcessor